src/entities/policy/policy.ts

```diff
@@ -1,36 +1,30 @@
 import { Duty } from "./duty";
 import { Permission } from "./permission";
 import { Prohibition } from "./prohibition";
-import { JsonLdId } from "./jsonld";
+import { JsonLdObject } from "../jsonld";
 
 export type PolicyType =
   | "set"
   | "offer"
   | "contract";
 
-export interface Policy {
+export interface PolicyInput {
   uid?: string;
   "@context"?: string;
-  "@type"?: {
-    "@policytype": PolicyType;
-  };
+  "@type"?: PolicyType;
   assignee?: string;
   assigner?: string;
   extensibleProperties?: object;
   inheritsFrom?: string;
   obligations?: Duty[];
-  permissions?: Permission[];
+  permission?: Permission[];
   prohibitions?: Prohibition[];
   target?: string;
 }
 
-export class PolicyDefinition extends JsonLdId {
-  createdAt?: number;
-  policy?: Policy;
-}
+export class Policy extends JsonLdObject {
 
-export interface PolicyDefinitionInput {
-  '@id'?: string;
-  id?: string;
-  policy: Policy;
+  get permissions(): JsonLdObject[] {
```

Comment:
Shall we keep naming alignment (i.e., `permission` v `permissions`)?

Reply:
so, ODRL asks for the singular, but tbh a method that returns a list in my opinion should be plural for readability.
Currently we have this "interface model" for building request objects that must follow the protocol 1:1, instead, on the response side we have this json-ld wrappers that permit to define a mode dev-friendly interface.
My take on this is that we could (should?) switch to the second approach for the input as well, but I know that js developers love that interface-objects stuff :)